### DISCUSSION TITLE
Verhandlungs Assistent im FOE Helper

### DISCUSSION THREAD

**Discussion by Bison0555** [mainIine/foe-helfer-extension#3071] [General]
Hallo bei mir gehen beim FOE Helper die Icons beim Verhandeln nicht hat das scjhon mal jemand gehabt und eine Lösung dafür?

**Comment by zarsys** [comment]
Bei mir auch gleiches Problem seit Update auf Weltbasis (Windows 8.1 Pro + Firefox).
Hatte woanders gleiches Problem schon mal gesehen, wurde durch Update auf Windows automaitsch 10 behoben.
Würde mich auch interessiere wir man dieses Problem ohne Windows Update lösen kann.
Habe alle Cookies gelöscht, Firefox und Add-ons neu installiert usw. - alles läuft überall 1A nur beim Verhandeln keine Icons!

![Firefox_Screenshot_2024-10-17T17-49-03 408Z](https://github.com/user-attachments/assets/525becec-5fe1-4253-a9b3-2feb8abbb8d6)

**Comment by Bison0555** [comment]
Leider bis jetzt keine Erkenntnisse. Zumal der Browser aktuell ist und das ja ein Browserspiel ist und die Windows Version eigentlich egal sein sollte. Zumal alles andere im Helper funktioniert.

**Comment by zarsys** [comment]
Ich habe unter Chrome auch keine Icons, alles andere funk. einwandfrei und zwar seit Jahren.

Der Firefox Debugger zeigt:

FOE Helper version 3.8.0.0 started. ID: {0668a87f-e3bb-4ed3-a096-e145c1e9f1d2} _main.js:2071:9
Mozilla/5.0 (Windows NT 6.3; Win64; x64; rv:109.0) Gecko/20100101 Firefox/115.0 _main.js:2072:9

**Alerts - Garbage collection failed** merged-3rd-party-109b844c.js:16:7623
**Error: No implementation for Extension communication found**

    sendExtMessage moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/web/_main/js/_main.js:1016
    getAll moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/web/alerts/js/alerts.js?v=3.8.0.0:222
    refresh moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/web/alerts/js/alerts.js?v=3.8.0.0:406
    garbage moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/web/alerts/js/alerts.js?v=3.8.0.0:368
    init moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/web/alerts/js/alerts.js?v=3.8.0.0:1591
    <anonymous> moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/web/_main/js/_main.js:291
    _proxyAction moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/foeproxy.js:396
    proxyAction moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/foeproxy.js:409
    xhrOnLoadHandlerExec moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/foeproxy.js:527
    xhrOnLoadHandler moz-extension://69154004-3395-48e4-bb5b-485d4e05780b/js/foeproxy.js:476

Ich habe mir den dazugehörigen Quellcode zu der garbage Funktion angeschaut (da wo Fehler produziert wird) - 
und folg. Zeilen gefunden:

		"try {
				await tmp.data.refresh();
				// TODO modify this to enable the display of alerts which expired while offline
				// one option ^ is to modify refresh"

"TODO" bedeutet - hier wird noch gearbeitet bzw. sollte noch irgendwas gemacht werden...

**Comment by Beelzebob6666** [comment]
#3054 Eure Browser sind veraltet und unfähig moderne Stylesheets zu lesen!

Wir werden auf Fortschritt verzichten, und in der nächsten Version den alten Stil wieder herstellen.

**Comment by zarsys** [comment]
Veraltet? Ich habe "115.16.1esr" aus Okt. 2024

**Reply by Beelzebob6666** [reply]
Das ist nur ein Sicherheitspatch gewesen, der eine alte FF Version vom Juli letzten Jahres für veraltete Betriebssysteme am Leben hält - die aktuelle Firefox version ist die 131.

**Comment by zarsys** [comment]
Das wusste niht nicht, Firefox sagte meine Version wäre "aktuell". Was micht aber wundert - es funktioniert alles und überall nur diese Icons nicht. Es sieht mehr nach einem kleinen Problem bei dem Verhandlungs Assistent aus, als nach einem globalen Problem mit Styles oder Browser Versionen... Sonst hätten wir überall gleiche Probleme und zwar massiv und nicht nur diese Kleinigkeit.

**Reply by Beelzebob6666** [reply]
Es liegt aber mit sehr hoher Sicherheit genau daran. An anderen Stellen wo wir das schon seit langem verwendet haben, fällt es wahrscheinlich nur nicht so auf, dass es nicht richtig formatiert ist.

https://developer.mozilla.org/en-US/docs/Web/CSS/Nesting_selector#browser_compatibility

**Comment by zarsys** [comment]
Könnt ihr nicht eine Zwischenlösung basteln anstatt "den alten Stil wieder herstellen"? Z.b. oben wo "Person 1", "Person 2" steht diese Güter im Klammern abgekürz auf 2-3 Bustabend ausschreiben? "Person 1 (Du)" Person 2 (We)" für Dunkle Energie bzw. Weltraumdaten? Dann weiß man auch ohne Icons bescheid...Bzw. unter "Person 1" noch eine kleine Zeile platzieren und die Güter als Text komplett stehen?
